background/services/enrichment/transactions.ts

```diff
@@ -0,0 +1,342 @@
+    (
+      await Promise.allSettled(
+        getDistinctRecipentAddressesFromERC20Logs(relevantTransferLogs).map(
+          async (address) =>
+            [
+              normalizeEVMAddress(address),
+              await enrichAddressOnNetwork(chainService, nameService, {
+                address,
+                network,
+              }),
+            ] as const
+        )
+      )
+    )
+      .filter(isFulfilledPromise)
+      .map(({ value }) => value)
+      .filter(([, annotation]) => isDefined(annotation))
+  )
+
+  const subannotations = tokenTransferLogs.flatMap<TransactionAnnotation>(
+    ({ contractAddress, amount, senderAddress, recipientAddress }) => {
+      // See if the address matches a fungible asset.
+      const matchingFungibleAsset = assets.find(
+        (asset): asset is SmartContractFungibleAsset =>
+          isSmartContractFungibleAsset(asset) &&
+          sameEVMAddress(asset.contractAddress, contractAddress)
+      )
+
+      if (!matchingFungibleAsset) {
+        return []
+      }
+
+      // Try to find a resolved annotation for the recipient and sender
+      const recipient =
+        annotationsByAddress[normalizeEVMAddress(recipientAddress)]
+      const sender = annotationsByAddress[normalizeEVMAddress(senderAddress)]
+
+      return [
+        {
+          type: "asset-transfer",
+          assetAmount: enrichAssetAmountWithDecimalValues(
+            {
+              asset: matchingFungibleAsset,
+              amount,
+            },
+            desiredDecimals
+          ),
+          sender,
+          recipient,
+          timestamp: resolvedTime,
+          blockTimestamp: block?.timestamp,
+        },
+      ]
+    }
+  )
+
+  return subannotations
+}
+
+/**
+ * Resolve an annotation for a partial transaction request, or a pending
+ * or mined transaction.
+ */
+export default async function resolveTransactionAnnotation(
+  chainService: ChainService,
+  indexingService: IndexingService,
+  nameService: NameService,
+  network: EVMNetwork,
+  transaction:
+    | AnyEVMTransaction
+    | (PartialTransactionRequestWithFrom & {
+        blockHash?: string
+      }),
+  desiredDecimals: number
+): Promise<TransactionAnnotation> {
+  // By default, annotate all requests as contract interactions
+  let txAnnotation: TransactionAnnotation = {
+    blockTimestamp: undefined,
+    timestamp: Date.now(),
+    type: "contract-interaction",
+    contractInfo: await enrichAddressOnNetwork(chainService, nameService, {
+      address: transaction.from,
+      network,
+    }),
+  }
+
+  let block: AnyEVMBlock | undefined
+
+  const {
+    assetAmount: { amount: baseAssetBalance },
+  } = await chainService.getLatestBaseAccountBalance({
+    address: transaction.from,
+    network,
+  })
+
+  const { gasLimit, blockHash } = transaction
+
+  const gasFee: bigint = isEIP1559TransactionRequest(transaction)
+    ? transaction?.maxFeePerGas ?? 0n * (gasLimit ?? 0n)
```

Comment:
we want to do `maxFeePerGas * gasLimit` right? so to fix this calculation it should be:
```suggestion
    ? (transaction?.maxFeePerGas ?? 0n) * (gasLimit ?? 0n)
```

Reply:
Right you are!